pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -411,30 +411,37 @@ func (p *PriorityQueue) AddUnschedulableIfNotPresent(pod *v1.Pod, podSchedulingC
 	if _, exists, _ := p.podBackoffQ.Get(pod); exists {
 		return fmt.Errorf("pod is already present in the backoffQ")
 	}
+
+	// Every unschedulable pod is subject to backoff timers.
+	p.backoffPod(pod)
+
+	// If there has been no move request since this pod failed scheduling, we can
+	// assume conditions have not changed and proceed normally.
 	if podSchedulingCycle > p.moveRequestCycle && isPodUnschedulable(pod) {
-		p.backoffPod(pod)
 		p.unschedulableQ.addOrUpdate(pod)
 		p.nominatedPods.add(pod, "")
 		return nil
 	}
 
-	// If a move request has been received and the pod is subject to backoff, move it to the BackoffQ.
+	// If a move request has been received and the pod is still waiting for its
+	// backoff timer to pass, move it to the BackoffQ.
 	if p.isPodBackingOff(pod) && isPodUnschedulable(pod) {
```

Comment:
Since we are backing pod off unconditionally, there is no point in checking `isPodBackingOff`. The function should always return true.

Reply:
good point.